core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -40,7 +42,7 @@
                            Collection<String> selectedColumns, ImmutableMap<String, String> options,
                            long fromSnapshotId, long toSnapshotId) {
     super(ops, table, null, schema, rowFilter, caseSensitive, colStats, selectedColumns, options);
-    Preconditions.checkArgument(fromSnapshotId != toSnapshotId, "fromSnapshotId and toSnapshotId cannot be the same");
+    validateSnapshotIds(table, fromSnapshotId, toSnapshotId);
```

Comment:
Moved snapshot id validations under the constructor, so that both api calls `DataTableScan.appendsBetween` and `IncrDataTableScan.appendsBetween` can be validated